### ISSUE TITLE
False positive when accessing a TypedDict in a list comprehension

### ISSUE THREAD

**Issue by fishikles** [python/mypy#19317]
**Bug Report**

When using a TypedDict in list comprehension 

**To Reproduce**

```python
from typing import TypedDict

class Data (TypedDict):
    field1:int
    field2:str
    field3:str

data:Data = {
    'field1':1,
    'field2':'string',
    'field3':'string'
}

stringValues = [value for value in ("field1", "field2", "field3") if type(data[value]) is str] # error here
```

**Expected Behavior**

Works as the non list comprehension verson:
```python
from typing import TypedDict

class Data (TypedDict):
    field1:int
    field2:str
    field3:str

data:Data = {
    'field1':1,
    'field2':'string',
    'field3':'string'
}
for value in ("field1", "field2", "field3"):
    if type(data[value]) is str:
        print(f"{value} is a string")
```

**Actual Behavior**

mypy complains about the typing of value in the list comprehension. Error reported is
```
error: TypedDict key must be a string literal; expected one of ("field1", "field2", "field3")  [literal-required]
```

**Your Environment**

- Mypy version used: mypy 1.16.0 (compiled: yes)
- Mypy command-line flags: NA
- Mypy configuration options from `mypy.ini` (and other config files): NA
- Python version used: 3.13.5

**Comment by brianschubert** [comment]
Something similar to #18014 would need to be done for comprehensions
